libs/content/editor/egui_editor/src/images.rs

```diff
@@ -49,60 +51,78 @@ pub fn calc(
 
                 result.map.insert(url.clone(), image_state.clone());
 
-                // download image
+                // fetch image
                 thread::spawn(move || {
                     let texture_manager = ctx.tex_manager();
 
-                    // use core for lb:// urls
-                    // todo: also handle relative paths
-                    let image_bytes = if let Some(stripped) = url.strip_prefix("lb://") {
-                        match Uuid::parse_str(stripped) {
-                            Ok(id) => match core.read_document(id) {
-                                Ok(bytes) => bytes,
-                                Err(e) => {
-                                    *image_state.lock().unwrap() =
-                                        ImageState::Failed(e.to_string());
-                                    ctx.request_repaint();
-                                    return;
+                    match (|| -> Result<TextureId, String> {
+                        // use core for lb:// urls
+                        // todo: also handle relative paths
+                        let maybe_lb_id = match url.strip_prefix("lb://") {
+                            Some(id) => Some(Uuid::parse_str(id).map_err(|e| e.to_string())?),
+                            None => None,
+                        };
+
+                        let image_bytes = if let Some(id) = maybe_lb_id {
+                            core.read_document(id).map_err(|e| e.to_string())?
+                        } else {
+                            download_image(&client, &url).map_err(|e| e.to_string())?
+                        };
+
+                        // convert lockbook drawings to images
+                        let image_bytes = if let Some(id) = maybe_lb_id {
+                            let file = core.get_file_by_id(id).map_err(|e| e.to_string())?;
+                            if file.name.ends_with(".svg") {
+                                // todo: check errors
+                                let tree = usvg::Tree::from_data(&image_bytes, &Default::default())
+                                    .map_err(|e| e.to_string())?;
+                                let tree = resvg::Tree::from_usvg(&tree);
+                                if let Some(content_area) = tree.content_area {
```

Comment:
nice, irrespective of where the user is zoomed in, you'd display the region that captures all elements in the canvas.
I didn't know about `tree.content_area`, cool, i thought you'd have to calculate that on your own.